### PR TITLE
Fixed-point:  `mul`, `div`, `pow`

### LINKED ISSUE #77: [Feature]: `div` for fixed-point math lib
### What is the feature you would like to see?

Implement fixed-point division.

### PR DESCRIPTION
**Pull request by immrsd**
Resolves #77, #87, #88

## Summary by CodeRabbit

* **New Features**
  * Added multiply, divide, and two power variants for SD29x9 and UD30x9; added one() helper constructors.
* **Bug Fixes / Reliability**
  * Improved overflow detection and tighter wrapping behavior for arithmetic operations.
* **Tests**
  * Greatly expanded test coverage for mul/div/pow edge cases, truncation, sign, and overflow scenarios (including expected abort checks).
* **Chores**
  * Updated .gitignore to ignore common Node.js artifacts.

### CODE REVIEW

File: math/fixed_point/sources/sd29x9/sd29x9.move [hunk 1]
```diff
@@ -83,21 +92,28 @@ public fun max(): SD29x9 {
...
+            min()
+        } else {
+            // The value is too large to be represented as a negative SD29x9
+            abort EOverflow
+        }
     } else {
-        SD29x9(x)
+        assert!(value <= MAX_POSITIVE_VALUE, EOverflow);
+        SD29x9(value)
     }
```
[the remark this answers is not in the document]
Comment (immrsd) [reply math/fixed_point/sources/sd29x9/sd29x9.move]: Reverted the changes to the `wrap` function

File: math/fixed_point/sources/sd29x9/sd29x9_base.move [hunk 2]
```diff
@@ -97,51 +98,103 @@ public fun is_zero(x: SD29x9): bool {
...
 
-/// Implements unary negation operation (-x) for SD29x9 type.
+/// Implements the checked multiplication operation (*) for `SD29x9` type.
+public fun mul(x: SD29x9, y: SD29x9): SD29x9 {
+    let x = decompose(x.unwrap());
+    let y = decompose(y.unwrap());
+    let neg = x.neg != y.neg;
+    let prod = (x.mag as u256) * (y.mag as u256);
+    let mag = (prod / SCALE_U256) as u128;
+    wrap_components(Components { neg, mag })
```
[the remark this answers is not in the document]
Comment (immrsd) [reply math/fixed_point/sources/sd29x9/sd29x9_base.move]: Not related to this PR

**Review by bidzyyys — APPROVED** [review]
LGTM
